fluid/documentation/CMakeLists.txt

```diff
@@ -0,0 +1,230 @@
+
+  # FIXME: This must also work with tarballs where git is not available.
+  #        For now we just ignore errors and set GIT_REVISION = "unknown".
+  #        In the future tarball/zip generation should create a file
+  #        that contains the git revision.
+
+  execute_process(COMMAND
+    git rev-parse --short=10 HEAD
+    OUTPUT_VARIABLE GIT_REVISION
+    OUTPUT_STRIP_TRAILING_WHITESPACE
+    WORKING_DIRECTORY ${FLTK_SOURCE_DIR}
+    ERROR_QUIET
+  )
+
+  # set to "'unknown'" if git is not available
+  if(GIT_REVISION STREQUAL "")
+    set(GIT_REVISION "'unknown'")
+  endif()
+
+  # Find "short" doxygen version if it was built from Git
+  # Note: this is still needed in CMake 3.12.0 but later CMake versions
+  #       (notably 3.25) remove the Git revision in 'DOXYGEN_VERSION'.
+  # Todo: Find the "first good" CMake version and remove this redundant
+  #       code once we require this as our minimal version and replace the
+  #       variable DOXYGEN_VERSION_SHORT with DOXYGEN_VERSION below.
+
+  if(DOXYGEN_FOUND)
+    # strip trailing git revision if doxygen was built from source
+    string(REGEX REPLACE " .*$" "" DOXYGEN_VERSION_SHORT ${DOXYGEN_VERSION})
+  endif(DOXYGEN_FOUND)
+
+  # configure copyright.dox (includes current year)
+  configure_file(
+    ${CMAKE_CURRENT_SOURCE_DIR}/copyright.dox.in
+    ${CMAKE_CURRENT_BINARY_DIR}/copyright.dox
+    @ONLY
+  )
+
+  # configure generated.dox (includes date and versions)
+  configure_file(
+    ${CMAKE_CURRENT_SOURCE_DIR}/generated.dox.in
+    ${CMAKE_CURRENT_BINARY_DIR}/generated.dox
+    @ONLY
+  )
+
+endif(FLTK_BUILD_FLUID_DOCS)
+
+#-------------------------------
+# build FLUID html documentation
+#-------------------------------
+
+if(FLTK_BUILD_FLUID_DOCS)
+
+  #list(APPEND DOCS html)
+
+  # generate Doxygen file "Doxyfile"
+
+  set(GENERATE_FLUID_HTML YES)
+  set(GENERATE_LATEX NO)
+  set(LATEX_HEADER   "")
+  set(DOXYFILE       "Doxyfile")
+  set(LOGFILE        "${CMAKE_CURRENT_BINARY_DIR}/${DOXYFILE}_error.log")
+
+  # configure Doxygen input file for HTML docs (Doxyfile.in)
+
+  configure_file(
+    ${CMAKE_CURRENT_SOURCE_DIR}/Doxyfile.in
+    ${CMAKE_CURRENT_BINARY_DIR}/${DOXYFILE}.in
+    @ONLY
+  )
+
+  # convert Doxyfile to used doxygen version
+
+  add_custom_command(
+    OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/${DOXYFILE}
+    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
+    COMMAND ${CMAKE_CURRENT_SOURCE_DIR}/convert_doxyfile
+            ${DOXYGEN_EXECUTABLE}
+            ${DOXYFILE}.in
+            ${DOXYFILE}
+            ${LOGFILE}
+    BYPRODUCTS ${LOGFILE}
+    COMMENT "Converting ${DOXYFILE} to doxygen version ${DOXYGEN_VERSION_SHORT}" VERBATIM
+  )
+
+  # generate screen shot using FLUID --autodoc target_dir
+  # generate HTML documentation
+
+  add_custom_target(fluid_docs
+    COMMAND echo fluid::fluid -scheme gtk+ --autodoc ${CMAKE_CURRENT_BINARY_DIR}/src/
+    COMMAND fluid::fluid -scheme gtk+ --autodoc ${CMAKE_CURRENT_BINARY_DIR}/src/
```

Comment:
s/fluid::fluid/fltk::fluid/ ???

Reply:
LOL, yes. Sorry, I will generate a bunch of commits until the workflow is well.

Reply:
> Sorry, I will generate a bunch of commits until the workflow is well.

That's what forks are good for. ;-)